### PR TITLE
Dataflow: Show existing connections

### PR DESCRIPTION
**Pull request by NgoEva**
- Restructure components
- Use grid layout to render nodes and connections
- Show existing connections based on selected source and destination LNode
- Component tests for Connections
- Redesign Dataflow Creation form
- Refactor component tests (remove unnecessary `awaits`, use `getByLabelText` instead of test-id)

Resolves:
- #158

### CODE REVIEW

File: packages/extensions/dataflow/src/lnode/dataflow-view.vue
Comment (NgoEva) [inline packages/extensions/dataflow/src/lnode/dataflow-view.vue:1]: Everything was extracted from the previous `dataflow-visualisation.vue` file

File: packages/extensions/dataflow/src/lnode/dataflow-view.test.ts
Comment (NgoEva) [inline packages/extensions/dataflow/src/lnode/dataflow-view.test.ts:1]: These were just moved from dataflow-visualisation.test.ts

Comment (NgoEva) [inline packages/extensions/dataflow/src/lnode/dataflow-view.test.ts:1]: These were just moved from `dataflow-visualisation.test.ts`

File: packages/extensions/dataflow/src/lnode/dataflow-view.vue
Comment (NgoEva) [inline packages/extensions/dataflow/src/lnode/dataflow-view.vue:1]: No new logic, everything was extracted from the previous `dataflow-visualisation.vue` file

File: packages/extensions/dataflow/src/lnode/dataflow-creation-form.test.ts [hunk 1]
```diff
@@ -104,45 +104,31 @@ describe('DataflowCreationForm', () => {
...
+		it('renders source and destination lnode label', () => {
 			const screen = render(DataflowCreationForm, {
 				props: {
 					sourceLNode: sendingLNode,
 					destinationLNode: receivingLNode,
 					isOpen: true,
 				},
 			})
-			await expect
-				.element(screen.getByText(getLNodeLabel(sendingLNode), { exact: true }))
```
Comment (NikoStu) [inline packages/extensions/dataflow/src/lnode/dataflow-creation-form.test.ts:141]: Is there a reason why you put 'sendingNode' in there as a variable and not as a string?

File: packages/extensions/dataflow/src/lnode/dataflow-visualisation.vue [hunk 2]
```diff
@@ -1,111 +1,155 @@
...
-	lNodes.value = await props.sdks.lnodeSDK.findAllEnrichedFromDB()
-}
+const sourceLNodeId = ref<string | undefined>()
+const destinationLNodeId = ref<string | undefined>()
 
-async function initConnections() {
-	if (!props.sdks) {
-		return
+const filteredConnections = computed(() => {
+	if (sourceLNodeId.value && destinationLNodeId.value) {
```
Comment (trusz) [inline packages/extensions/dataflow/src/lnode/dataflow-visualisation.vue:136]: nit: an early-return could imprive the readability a bit

File: packages/extensions/dataflow/src/lnode/dataflow-view.test.ts [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+				dataAttributeSpecifications: [
+					{ id: 'das2', name: 'SpecAttr2', desc: 'dummy', dataObjectSpecificationId: 'dos2' },
+				],
+				lNodeId: '2',
+			},
+		],
+	},
+]
+
+describe('DataflowView: Dataflow creation', () => {
```
Comment (trusz) [inline packages/extensions/dataflow/src/lnode/dataflow-view.test.ts:84]: The test are nice and readable :)

**Review by trusz — APPROVED** [review]

File: packages/extensions/dataflow/src/lnode/dataflow-view.vue [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+function onSourceLNodeChange(newLNodeId?: string) {
+	sourceLNode.value = getActiveLNodeById(newLNodeId)
+}
+
+function onDestinationLNodeChange(newLNodeId?: string) {
+	destinationLNode.value = getActiveLNodeById(newLNodeId)
+}
+
+function getActiveLNodeById(id: string | undefined) {
+	return lnodes.value.find((ln) => ln.id === id) ?? null
```
[the remark this answers is not in the document]
Comment (NgoEva) [reply packages/extensions/dataflow/src/lnode/dataflow-view.vue:79]: It doesn't matter, I did not add this, this was just moved from another file. But I can change it to undefined, yes.

File: packages/extensions/dataflow/src/app.vue [hunk 5]
```diff
@@ -72,17 +69,6 @@ async function initSDKs(newActiveFile: string) {
 }
 </script>
 
-<style scoped>
-/*
-	We import it here and not on the top so it is scoped and does not affect
-	the rest of the ui
-*/
+<style>
```
[the remark this answers is not in the document]
Comment (NgoEva) [reply packages/extensions/dataflow/src/app.vue:72]: Oh yeah, I did not look into this. I only thought about importing it once here on top-level for my extension views, but did not checkout the overall implications.

File: packages/extensions/dataflow/src/lnode/dataflow-visualisation.vue [hunk 6]
```diff
@@ -1,111 +1,155 @@
...
+		<div
+			class="col-start-4 col-span-2 row-span-full bg-(--color-ocean-gray-50) relative rounded-3xl -z-1 border-(--color-ocean-gray-100) border-3 min-h-[500px]"
+		></div>
+
+		<!-- Right LNode Select -->
+		<select
+			:value="destinationLNodeId ?? ''"
+			@change="(e) => onDestinationLNodeSelect((e.target as HTMLSelectElement).value)"
+			data-testid="select-destination-lnode"
+			class="col-start-4 col-span-2 self-center justify-self-center row-start-1 text-2xl text-center"
```
[the remark this answers is not in the document]
Comment (NgoEva) [reply packages/extensions/dataflow/src/lnode/dataflow-visualisation.vue:35]: I thought so too hahah. In the design the label for the node was that big, but I couldn't figure out a way how to make the select small but just the selected value label big :D

**Comment by NgoEva** [comment]
> One thing is puzzling me: can/should we allow two times the same LNode? I guess the connections are a edge case because of the handmade test file? 😅
> 
> Bildschirmfoto 2025-08-27 um 08 58 11

I noticed the edge case too. I can prevent selecting the same. But I did not know that the test file has connections to the same LNode?! that is really weird o.O
